src/services/pfs/clients/__tests__/FileSet.test.ts

```diff
@@ -0,0 +1,99 @@
+import path from 'path';
+
+import client from '../../../../client';
+
+describe('FileSet', () => {
+  afterAll(async () => {
+    const pachClient = client({ssl: false, pachdAddress: 'localhost:30650'});
+    const pfs = pachClient.pfs();
+    await pfs.deleteAll();
+  });
+
+  const createSandbox = async (name: string) => {
+    const pachClient = client({ssl: false, pachdAddress: 'localhost:30650'});
+    const pfs = pachClient.pfs();
+    await pfs.deleteAll();
+    await pfs.createRepo({repo: {name}});
+
+    return pachClient;
+  };
+
+  describe('putFileFromURL', () => {
+    it('should add a file from a URL to a repo', async () => {
+      const client = await createSandbox('putFileFromURL');
+
+      const fileClient = await client.pfs().fileSet();
+      const fileSetId = await fileClient
+        .putFileFromURL('at-at.png', 'http://imgur.com/8MN9Kg0.png')
+        .end();
+
+      const commit = await client.pfs().startCommit({
+        branch: {name: 'master', repo: {name: 'putFileFromURL'}},
+      });
+
+      await client.pfs().addFileSet({fileSetId, commit});
+      await client.pfs().finishCommit({commit});
```

Comment:
(nit): Could we create some sort of helper function that aggregates this common set of calls?

```
await client
  .pfs()
  .fileSet()
  .publishCommitFromURL(// ... file urls)
```

I don't think this is necessary, and exposing the individual operations gives us flexibility, but I bet the usage in Console will mostly be folks adding a group of files to a commit and pushing it.

Reply:
We could definitely do that. I'm a little torn between whether that function should live in `pfs` or `FileSet` due to all of those calls coming from `pfs` and not the `FileSet` stream